src/solvers/dgsem_p4est/dg_parallel.jl

```diff
@@ -5,13 +5,12 @@
 @muladd begin
 #! format: noindent
 
-mutable struct P4estMPICache{BufferType <: DenseVector,
-                             VecInt <: DenseVector{<:Integer}}
+mutable struct P4estMPICache{uEltype}
     mpi_neighbor_ranks::Vector{Int}
-    mpi_neighbor_interfaces::VecOfArrays{VecInt}
-    mpi_neighbor_mortars::VecOfArrays{VecInt}
-    mpi_send_buffers::VecOfArrays{BufferType}
-    mpi_recv_buffers::VecOfArrays{BufferType}
+    mpi_neighbor_interfaces::VectorOfArray{Int, 2, Vector{Vector{Int}}}
+    mpi_neighbor_mortars::VectorOfArray{Int, 2, Vector{Vector{Int}}}
+    mpi_send_buffers::VectorOfArray{uEltype, 2, Vector{Vector{uEltype}}}
+    mpi_recv_buffers::VectorOfArray{uEltype, 2, Vector{Vector{uEltype}}}
```

Comment:
This seems counter what I expect. For GPU support the send and receive buffers likely need to be CuArrays.